### PR TITLE
Adds /teams route

### PR DESCRIPTION
**Pull request by Ricky12Awesome**
Implements #331

### CODE REVIEW

**Comment by wafflecoffee** [comment]
You're going to need to run `cargo sqlx prepare` for it to compile with GitHub Actions.

**Review by wafflecoffee — APPROVED** [review]

File: src/routes/teams.rs [hunk 1]
```diff
@@ -101,6 +101,65 @@ pub async fn team_members_get(
...
+    let teams_data =
+        TeamMember::get_from_team_full_many(team_ids, &**pool).await?;
+
+    let teams_groups = teams_data.into_iter().group_by(|data| data.team_id.0);
+
+    let mut teams: Vec<Vec<crate::models::teams::TeamMember>> = vec![];
+
+    for (id, member_data) in &teams_groups {
+        let current_user =
+            get_user_from_headers(req.headers(), &**pool).await.ok();
```
Comment (Geometrically) [inline src/routes/teams.rs:131]: why is this done every iteration?

File: src/routes/teams.rs [hunk 2]
```diff
@@ -101,6 +101,65 @@ pub async fn team_members_get(
...
+
+        if let Some(user) = current_user {
+            let team_member = TeamMember::get_from_user_id(
+                crate::database::models::ids::TeamId(id),
+                user.id.into(),
+                &**pool,
+            )
+            .await
+            .map_err(ApiError::Database)?;
```
Comment (Geometrically) [inline src/routes/teams.rs:140]: Is it possible to batch this with raw SQL- querying the database in a for loop is a bad idea and can be abused

**Review by Geometrically — CHANGES_REQUESTED** [review]
This PR is querying the database really inefficiently.

It's not your fault- but avoid mass queries in for loops

File: src/routes/teams.rs (hunk 1 above)
Comment (Ricky12Awesome) [reply src/routes/teams.rs:131]: Can put that out of the loop

File: src/routes/teams.rs (hunk 2 above)
Comment (Ricky12Awesome) [reply src/routes/teams.rs:140]: Should be possible, working on that now

**Review by Geometrically — COMMENTED** [review]
LGTM- does this hide member permissions (returns None for permissions)

**Comment by Ricky12Awesome** [comment]
Yes, when I call the endpoint, it displays null for permissions, with `Authorization` set, it only shows for the teams the user is apart of

**Review by Geometrically — APPROVED** [review]

**Pull request merged**